### PR TITLE
Add info on --rerun task option to CLI docs

### PR DESCRIPTION
**Pull request by abstratt**
Doc changes for #20895.

Issue: #9166 

Related PR: #20895 

### Context

Please see #9166 and #20895. This is a doc-only change set.

### Gradle Core Team Checklist
- [ ] Verify documentation

### CODE REVIEW

**Comment by abstratt** [comment]
@bot-gradle help

**Comment by abstratt** [comment]
@bot-gradle SF QF

**Comment by abstratt** [comment]
@bot-gradle clean

**Comment by abstratt** [comment]
@bot-gradle test SC QF

File: subprojects/docs/src/docs/userguide/reference/command_line_interface.adoc [hunk 1]
```diff
@@ -598,6 +600,23 @@ Set JVM arguments.
 `-Dorg.gradle.java.home`::
 Set JDK home dir.
 
+[[sec:task_options]]
+== Task options
```
Comment (eskatos) [inline subprojects/docs/src/docs/userguide/reference/command_line_interface.adoc:604]: This should somehow link to https://docs.gradle.org/current/userguide/custom_tasks.html#sec:declaring_and_using_command_line_options

**Comment by eskatos** [comment]
@bot-gradle help

**Comment by eskatos** [comment]
@bot-gradle test BD

**Comment by eskatos** [comment]
Preview available at https://builds.gradle.org/repository/download/Gradle_Master_Check_BuildDistributions/52866120:id/distributions/gradle-7.6-docs.zip!/gradle-7.6-20220609235315%2B0000/docs/userguide/command_line_interface.html#sec:task_options

**Comment by abstratt** [comment]
@bot-gradle test BD

**Comment by abstratt** [comment]
https://builds.gradle.org/repository/download/Gradle_Master_Check_BuildDistributions/52872299:id/distributions/gradle-7.6-docs.zip!/gradle-7.6-20220610110038%2B0000/docs/userguide/command_line_interface.html#sec:task_options

@eskatos What do you think?

**Review by eskatos — APPROVED** [review]
:+1:

**Comment by abstratt** [comment]
@bot-gradle test and merge

**Pull request merged**